### PR TITLE
TKX-3228 Allow getting prices for token symbols instead of coingecko ids

### PR DESCRIPTION
**Pull request by emanuelgaspar**

### CODE REVIEW

File: src/Trakx.CoinGecko.ApiClient/Clients/CoinGeckoClient_Public.cs [hunk 1]
```diff
@@ -71,20 +70,9 @@ public async Task<MultiplePrices> GetAllPrices(
...
-        (var baseIds, var quoteIds) = GetIdsForPriceQuery(ids, vsCurrencies, supportedCurrencies);
-
-        var response = await _simpleClient
-            .PriceAsync(baseIds, quoteIds, cancellationToken: cancellationToken)
-            .ConfigureAwait(false);
-
-        if (Logger.IsEnabled(LogLevel.Debug))
-        {
-            Logger.LogDebug("Received latest price {PriceAsyncResponse}", JsonSerializer.Serialize(response));
-        }
```
Comment (emanuelgaspar) [inline src/Trakx.CoinGecko.ApiClient/Clients/CoinGeckoClient_Public.cs:87]: Refactored to `GetAllPricesInternal`.

File: src/Trakx.CoinGecko.ApiClient/Clients/CoinGeckoClient_Internal.cs [hunk 2]
```diff
@@ -218,6 +238,8 @@ private static (string BaseIds, string QuoteIds) GetIdsForPriceQuery(
 
         vsCurrencies ??= [];
 
+        var supportedQuoteCurrencies = await GetSupportedQuoteCurrencies(cancellationToken);
```
Comment (emanuelgaspar) [inline src/Trakx.CoinGecko.ApiClient/Clients/CoinGeckoClient_Internal.cs:247]: This was being done by each caller, might as well move inside the common method.

File: src/Trakx.CoinGecko.ApiClient.Tests/Unit/CoinGeckoClientTests.cs [hunk 3]
```diff
@@ -288,123 +291,25 @@ public async Task GetMarketRank_should_reuse_previous_results_when_called_with_l
         updatedApiCalls.Should().Be(initialApiCalls);
     }
 
-    private void ConfigurePriceAsync(string _coin, string currency, decimal coinPrice, decimal currencyPrice)
```
Comment (emanuelgaspar) [inline src/Trakx.CoinGecko.ApiClient.Tests/Unit/CoinGeckoClientTests.cs:291]: Private methods moved to separate file.

File: src/Trakx.CoinGecko.ApiClient/Clients/CoinGeckoClient_Internal.cs
Comment (joaofbvieira) [inline src/Trakx.CoinGecko.ApiClient/Clients/CoinGeckoClient_Internal.cs:1]: Not in the context of this PR, but this class is quite big and have some important private methods on it like `GetSymbolToCoinGeckoIdMapInternal` which contains some relevante business rules that could be tested in a separate class/unit test (e.g. `CoinGeckoService`).

File: src/Trakx.CoinGecko.ApiClient/Clients/CoinGeckoClient_Public.cs [hunk 4]
```diff
@@ -53,8 +52,8 @@ public CoinGeckoClient(
         string quoteCurrencyId = Constants.UsdCoin,
         CancellationToken cancellationToken = default)
     {
-        if(coinGeckoId.IsNullOrWhiteSpace()) throw new ArgumentException($"{nameof(coinGeckoId)} cannot be null or empty");
-        if(quoteCurrencyId.IsNullOrWhiteSpace()) throw new ArgumentException($"{nameof(quoteCurrencyId)} cannot be null or empty");
+        if (coinGeckoId.IsNullOrWhiteSpace()) throw new ArgumentException($"{nameof(coinGeckoId)} cannot be null or empty");
+        if (quoteCurrencyId.IsNullOrWhiteSpace()) throw new ArgumentException($"{nameof(quoteCurrencyId)} cannot be null or empty");
```
Comment (joaofbvieira) [inline src/Trakx.CoinGecko.ApiClient/Clients/CoinGeckoClient_Public.cs]: Could this use `ArgumentException.ThrowIfNullOrWhiteSpace` instead?

**Review by joaofbvieira — DISMISSED** [review]

File: src/Trakx.CoinGecko.ApiClient/Clients/CoinGeckoClient_Public.cs (hunk 4 above)
Comment (emanuelgaspar) [reply src/Trakx.CoinGecko.ApiClient/Clients/CoinGeckoClient_Public.cs]: Good catch, thanks, done:
https://github.com/trakx/coingecko-api-client/pull/252/commits/c9a9f8b5d2633f4baa5ab7211be008d359086d9c

File: src/Trakx.CoinGecko.ApiClient/Clients/CoinGeckoClient_Internal.cs
Comment (emanuelgaspar) [reply src/Trakx.CoinGecko.ApiClient/Clients/CoinGeckoClient_Internal.cs:1]: Thanks, it's a valid concern.
I still have some unit tests to complete for the new features (left as TODO in this PR).
I will be sure to clean up the test files in the next PR.

**Review by joaofbvieira — APPROVED** [review]

**Pull request merged**